src/pages/_app.js

```diff
@@ -5,27 +5,26 @@ import * as React from 'react'
 import PropTypes from 'prop-types'
 import Head from 'next/head'
 import { CacheProvider } from '@emotion/react'
-import { ThemeProvider } from '@mui/material/styles'
-import { CssBaseline } from '@mui/material'
-import { settings as remoteConfig } from '~/api/__mock__'
-import { CentraProvider, GlobalProvider, I18nProvider, RemoteConfigProvider } from '~/context'
-import createEmotionCache from '~/utils/createEmotionCache'
-import theme from '~/utils/theme.light'
-import AppBase from '~/containers/App'
+import { settings } from '~/api/__mock__'
+import { createEmotionCache } from '~/utils'
+import { RootProvider } from '~/context'
+import * as layoutVariants from '~/layouts'
 
 // Client-side cache, shared for the whole session of the user in the browser.
 const clientSideEmotionCache = createEmotionCache()
 
 function App(props) {
   const {
-    cmsProps,
     Component,
     defaultLocale,
     emotionCache = clientSideEmotionCache,
     locale,
-    pageProps,
+    pageProps: pagePropsProp,
   } = props
 
+  const { headerColor, headerMode, layout, page, ...pageProps } = pagePropsProp
+  const LayoutComponent = layout ? layoutVariants[layout] : layoutVariants.App
+
```

Comment:
Whoa! This looks like a cool idea! 
Is it enabling different layout structures for blocks..?